src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -0,0 +1,222 @@
+
+    __all__ = ("record", "resolve", "reverse_resolve", "bulk_record")
+
+    def _record_many_strings(self, keys: Sequence[str]) -> Mapping[str, int]:
+        """
+        Bulk create db records for org_id:string pairs. As noted below, bulk_create with
+        ignore_conflicts=True returns objects that don't have the pk set on them (aka the `id`
+        that we need), so that's why we use `_get_db_ids_and_set_cache` to look up the records
+        and return a mapping of org_id:string -> id:
+            {
+                "1:release": 3,
+                "2:v1": 4
+            }
+        """
+        records = []
+        for key in keys:
+            organization_id, string = key.split(":")
+            records.append(StringIndexerTable(organization_id=int(organization_id), string=string))
+
+        with metrics.timer("sentry_metrics.indexer.pg_bulk_create"):
+            # We use `ignore_conflicts=True` here to avoid race conditions where metric indexer
+            # records might have be created between when we queried in `bulk_record` and the
+            # attempt to create the rows down below.
+            StringIndexerTable.objects.bulk_create(records, ignore_conflicts=True)
+
+        # We re-query the database to fetch the rows (which should all exist point at this point),
+        # in addition to setting the results in the cache
+        results = self._get_db_ids_and_set_cache(keys)
+        # we _just_ created this records, there is no reason for results to be None
+        assert results
+        return results
+
+    def _get_db_ids_and_set_cache(self, keys: Sequence[str]) -> Union[Mapping[str, int], None]:
+        """
+        Query the db for the org_id and string pairs given by the keys (e.g "1:release")
+
+        If we find no records out of the entire list of keys, return None. Otherwise, we
+        return a mapping of org_id:string key -> id:
+            {
+                "1:release": 3,
+                "2:v1": 4
+            }
+
+        When setting values in the cache we need to cache both ways:
+            * "org_id:string" -> id
+            * id -> "string"
+        The id doesn't need the org_id as part of the cache key because it's unique
+        across the StringIndexerTable.
+        """
+        conditions = []
+        for key in keys:
+            organization_id, string = key.split(":")
+            conditions.append(Q(organization_id=int(organization_id), string=string))
+
+        query_statement = reduce(or_, conditions)
+
+        db_results = {}
+        reverse_db_results = {}
+        db_objs = StringIndexerTable.objects.filter(query_statement)
+        if not db_objs:
+            return None
+
+        for db_obj in db_objs:
+            key = f"{db_obj.organization_id}:{db_obj.string}"
+
+            db_results[key] = db_obj.id
+            reverse_db_results[db_obj.id] = db_obj.string
+
+        indexer_cache.set_many(db_results)
+        indexer_cache.set_many(reverse_db_results)
```

Comment:
I don't think we need to add the reverse mapping to the cache eagerly. 
The two use cases (direct and reverse mappings) are used at very different volume. The reverse mapping is only for querying which happen a few times per second and probably concentrated on a small number of ids. 
The direct mapping is checked hundreds of times per second instead). You can just let the query pipeline take care of that and you save one roundtrip to memcache in the indexer.